### PR TITLE
refactor: rename TuiHintModule to TuiHint

### PR DESCRIPTION
**Pull request by splincode**

### CODE REVIEW

File: projects/demo/src/modules/pipes/field-error/field-error.module.ts [hunk 1]
```diff
@@ -40,14 +40,14 @@ import {ExampleTuiFieldErrorPipeComponent} from './field-error.component';
...
-        TuiHintModule,
+        TuiHint,
         TuiInputModule,
         TuiButtonDirective,
         TuiInputPhoneModule,
         TuiInputNumberModule,
         TuiLinkDirective,
         TuiTable,
-        TuiHintModule,
+        TuiHint,
```
Comment (nsbarsukov) [inline projects/demo/src/modules/pipes/field-error/field-error.module.ts:50]: Remove duplicated import

Comment (splincode) [reply projects/demo/src/modules/pipes/field-error/field-error.module.ts]: fixed

File: projects/core/directives/hint/hint-position.directive.ts [hunk 2]
```diff
@@ -18,6 +18,7 @@ const TOP = 0;
 const LEFT = 1;
 
 @Directive({
+    standalone: true,
     selector: '[tuiHint]:not(ng-container):not(ng-template)',
```
Comment (waterplea) [inline projects/core/directives/hint/hint-position.directive.ts:22]: We probably need to move it to `hostDirectives` of `TuiHintDirective` and make `:is(never)` selector so it does not add noise for autoimports.

File: projects/core/directives/hint/hint-hover.directive.ts [hunk 3]
```diff
@@ -20,6 +20,7 @@ import type {TuiHintOptions} from './hint-options.directive';
 import {TUI_HINT_OPTIONS} from './hint-options.directive';
 
 @Directive({
+    standalone: true,
     selector: '[tuiHint]:not(ng-container):not(ng-template)',
```
[the remark this answers is not in the document]
Comment (splincode) [reply projects/core/directives/hint/hint-hover.directive.ts:24]: Can I use a separate PR?

File: projects/core/directives/hint/hint-driver.directive.ts [hunk 4]
```diff
@@ -2,6 +2,7 @@ import {Directive} from '@angular/core';
 import {TuiDriverDirective} from '@taiga-ui/core/classes';
 
 @Directive({
+    standalone: true,
     selector: '[tuiHint]:not(ng-container):not(ng-template)',
```
[the remark this answers is not in the document]
Comment (splincode) [reply projects/core/directives/hint/hint-driver.directive.ts:6]: Can I use a separate PR?

File: projects/core/directives/hint/hint-position.directive.ts (hunk 2 above)
Comment (splincode) [reply projects/core/directives/hint/hint-position.directive.ts:22]: Can I use a separate PR?

**Review by waterplea — APPROVED** [review]

**Review by nsbarsukov — APPROVED** [review]

**Pull request merged**